thunder/core/proxies.py

```diff
@@ -71,17 +71,6 @@ def unvariableify(x: Any) -> Any:
     return x
 
 
-# Tries to register a proxy name and returns a boolean indicating success
-def register_proxy_name(name: None | str = None):
```

Comment:
What happens if two proxies share a trace and a name? Wouldn't the Python program use the same name to refer to both of them?

Reply:
Yes it would. When two proxies with the same name are used in a trace the Python program is generated as prescribed by the order of bound symbols:
```py
In [1]: from thunder.core.devices import cpu

In [2]: from thunder.core.dtypes import float32

In [3]: from thunder.core.proxies import TensorProxy

In [4]: p_test0 = TensorProxy(name="test", shape=(1,), device=cpu, dtype=float32)

In [5]: p_test1 = TensorProxy(name="test", shape=(1,), device=cpu, dtype=float32)

In [6]: a = TensorProxy(name="name", shape=(1,), device=cpu, dtype=float32)

In [7]: from thunder.core.prims import exp

In [8]: b1 = exp.bind(a, output=p_test1)

In [9]: b0 = exp.bind(a, output=p_test0)

In [10]: bsyms = [b0, b1]

In [11]: from thunder.core.trace import TraceCtx

In [12]: t = TraceCtx(None)

In [13]: t.bound_symbols = bsyms

In [14]: t
Out[14]: 
import thunder
import thunder.core.prims as prims
import torch
from thunder.executors.torchex import no_autocast

@torch.no_grad()
@no_autocast
# No signature available
  test = prims.exp(name)  # test: "cpu f32[1]"
  test = prims.exp(name)  # test: "cpu f32[1]"
```

Reply:
Isn't it a problem if names can collide like that?